p2p/src/main/kotlin/maru/p2p/messages/StatusMessageSerDe.kt

```diff
@@ -8,27 +8,47 @@
  */
 package maru.p2p.messages
 
-import maru.p2p.Message
+import maru.p2p.RequestMessage
+import maru.p2p.ResponseMessage
 import maru.p2p.RpcMessageType
 import maru.p2p.Version
 import maru.serialization.rlp.RLPSerDe
 import org.hyperledger.besu.ethereum.rlp.RLPInput
 import org.hyperledger.besu.ethereum.rlp.RLPOutput
 
-class StatusMessageSerDe(
-  private val statusSerDe: RLPSerDe<Status>,
-) : RLPSerDe<Message<Status, RpcMessageType>> {
+class StatusResponseMessageSerDe(
```

Comment:
Is it possible to have a common template class `StatusMessageSerDe` to replace `StatusResponseMessageSerDe` and `StatusRequestMessageSerDe` as their implementations look identical to me

Reply:
I used delegation in my [PR](https://github.com/Consensys/maru/pull/432/files#diff-83e5057287276c6571904c5ba91b6c8ce8e5141de62504eb56a3a20757b32bc3R20) instead